src/sui/sui.service.ts

```diff
@@ -11,6 +12,175 @@ import { Collateral, Debt } from 'src/obligation/obligation.schema';
+  public SUI_PAGE_LIMIT = Number(process.env.PAGE_LIMIT) || 1;
+
+  private API_URL = process.env.API_URL || 'https://sui.api.scallop.io/';
+  private API_KEY = process.env.API_KEY || 'scalloptestapikey';
+  private ADDRESSES_ID = process.env.ADDRESSES_ID || '6462a088a7ace142bb6d7e9b';
+  private NETWORK = process.env.NETWORK || 'testnet';
+
+  private MARKET_ID = process.env.MARKET_ID;
+  private PROTOCOL_ID = process.env.PROTOCOL_ID;
+  private INIT_PROTOCOL_ID =
+    '0xefe8b36d5b2e43728cc323298626b83177803521d195cfb11e15b910e892fddf';
+
+  private _addresses = undefined;
+  private _protocolId = undefined;
+  private _marketId = undefined;
+
+  // event id
+  private _obligationCreatedEventId = undefined;
+  private _collateralDepositEventId = undefined;
+  private _collateralWithdrawEventId = undefined;
+  private _borrowEventId = undefined;
+  private _repayEventId = undefined;
+  private _liquidateEventId = undefined;
+  private _flashloanBorrowEventId = undefined;
+  private _flashloanRepayEventId = undefined;
+
+  private _mintEventId = undefined;
+  private _redeemEventId = undefined;
+
+  private async fetchAddressesFromAPI() {
+    let addresses = undefined;
+    try {
+      const response = await axios.get(
+        this.API_URL + 'addresses/' + this.ADDRESSES_ID,
+        {
+          headers: {
+            'api-key': this.API_KEY,
+          },
+        },
+      );
+      addresses = response.data;
+
+      console.log('[Addresses]: fetchAddressesFromAPI() success');
+    } catch (e) {
+      console.error('Error caught while fetchAddressesFromAPI() ', e);
+    }
+    return addresses;
+  }
+
+  private async getAddresses() {
+    if (!this._addresses) {
+      this._addresses = await this.fetchAddressesFromAPI();
+    }
+    return this._addresses;
+  }
+
+  public async getMarketId() {
+    if (!this._marketId) {
+      if (this.MARKET_ID) {
+        this._marketId = this.MARKET_ID;
+      } else {
+        const address = await this.getAddresses();
+        if (this._addresses) {
+          this._marketId = address[this.NETWORK]['core']['market'];
+        }
+      }
+      console.log(`[Market-Id]: ${this._marketId}`);
+    }
+    return this._marketId;
+  }
+
+  private async getProtocolId() {
+    if (!this._protocolId) {
+      if (this.PROTOCOL_ID) {
+        this._protocolId = this.PROTOCOL_ID;
+      } else {
+        const address = await this.getAddresses();
+        if (this._addresses) {
+          this._protocolId =
+            address[this.NETWORK]['core']['packages']['protocol']['id'];
+        }
+      }
+      console.log(`[Protocol-Id]: ${this._protocolId}`);
+    }
+    return this._protocolId;
+  }
+
+  public async getObligationCreatedEventId() {
+    if (!this._obligationCreatedEventId) {
+      const protocol = await this.getProtocolId();
+      this._obligationCreatedEventId = `${protocol}::open_obligation::ObligationCreatedEvent`;
```

Comment:
The type of the event or the type of an object will stick to the `{{INITIAL_PROTOCOL_ID}}::any_module::any_type`